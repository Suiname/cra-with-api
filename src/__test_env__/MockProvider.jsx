import React from 'react';
import { Provider } from 'react-redux';
import createStore from '../store';

const MockProvider = ({ children }) => (
  <Provider store={createStore()}>{children}</Provider>
);

export default MockProvider;
