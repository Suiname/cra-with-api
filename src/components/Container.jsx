import React, { Component } from 'react';
import { connect } from 'react-redux';
import { loadMessage } from '../actions/example';

export class Container extends Component {

  componentDidMount() {
    if (!this.props.message) {
      this.props.loadMessage();
    }
  }

  render() {
    return (
      <div>
        {this.props.message}
      </div>
    );
  } 
}
    
const mapStateToProps = ({ example }) => ({
  message: example.message,
});

export default connect(mapStateToProps, { loadMessage })(Container);
