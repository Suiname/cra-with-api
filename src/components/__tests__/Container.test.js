import React from 'react';
import renderer from 'react-test-renderer';
import MockProvider from '../../__test_env__/MockProvider';
import { Container } from '../Container';

const loadMessage = jest.fn();

test('renders', () => {
  const component = renderer.create(
    <MockProvider>
      <Container loadMessage={loadMessage} />
    </MockProvider>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders message', () => {
  const component = renderer.create(
    <MockProvider>
      <Container message="Test message" />
    </MockProvider>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
