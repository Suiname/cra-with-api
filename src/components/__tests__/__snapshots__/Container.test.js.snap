exports[`test renders 1`] = `<div />`;

exports[`test renders message 1`] = `
<div>
  Test message
</div>
`;
